crates/services/txpool_v2/src/pool.rs

```diff
@@ -354,7 +354,8 @@ where
 
     /// Remove transaction but keep its dependents.
     /// The dependents become executables.
-    pub fn remove_transaction(&mut self, tx_ids: Vec<TxId>) {
+    pub fn remove_transactions(&mut self, tx_ids: Vec<TxId>) -> Vec<ArcPoolTx> {
```

Comment:
can this accept an iterator for `tx_ids` too please?
```suggestion
    pub fn remove_transactions(&mut self, tx_ids: impl Iterator<Item = TxId>) -> Vec<ArcPoolTx> {
```

perhaps you might want to return an Iterator for the `ArcPoolTx` too :)

Reply:
I find it useful to be able to have the len for the parameter to pre-alloc the result. And I don't really see how to make the result iterator (maybe with a lot of combinators) but it will resolve the pre-alloc problem so if you have an idea feel free to commit here

Reply:
think you can use size_hint() right?

Reply:
Ok, I added for the paramter but couldn't do it for return because of lifetimes etc and I think given the fact that we have to bond lifetime of self on the iteraztor returned it's less flexible so I will let the vec.